index.html

```diff
@@ -296,141 +292,176 @@
-            case "A": // A period
-            case "B": // B period
-            case "C": // C period
-            case "D": // D period
-            case "E": // E period
-            case "F": // F period
-            case "G": // G period
-            case "H": // H period
-                for (var k = 0; k < schedule["classes"].length; k++) {
-                    if (schedule["classes"][k]["period"] == days[0][day][school][currentSlot]["period"]) {
-                        if (schedule["classes"][k]["name"] != "Free Period") {
-                            var teacherName = schedule["classes"][k]["teacher"]; // Teacher's fullname
-                            var teacherLast = teacherName.split(" ")[1]; // Teacher's last name
-                            scheduleObj = { name: schedule["classes"][k]["name"],
-                                teacher: teacherName,
-                                teacherLastName: teacherLast,
-                                room: schedule["classes"][k]["room"],
-                                period: days[0][day][school][currentSlot]["period"],
-                                time: ""}
+    return day
+  }
 
-                        } else {
-                            scheduleObj = {name: "Free Period",
-                                teacher: "",
-                                teacherLastName: "",
-                                room: "",
-                                period: days[0][day][school][currentSlot]["period"],
-                                time: ""};
-                        }
-                    }
-                }
-                break;
+  // Generate basic name, teacher, room, time, and period info for a class
+  function renderClassBase(schedule, school, day, currentSlot) {
+    switch(days[0][day][school][currentSlot]["period"]) { // Falling through intended
+      case "0": // 0 Period
+      case "A": // A period
+      case "B": // B period
+      case "C": // C period
+      case "D": // D period
+      case "E": // E period
+      case "F": // F period
+      case "G": // G period
+      case "H": // H period
+        for (var k = 0; k < schedule["classes"].length; k++) {
+          if (schedule["classes"][k]["period"] == days[0][day][school][currentSlot]["period"]) {
+            if (schedule["classes"][k]["name"] != "Free Period") {
+              var teacherName = schedule["classes"][k]["teacher"]; // Teacher's fullname
+              var teacherLast = teacherName.split(" ")[1]; // Teacher's last name
+              scheduleObj = { name: schedule["classes"][k]["name"],
+                teacher: teacherName,
+                teacherLastName: teacherLast,
+                room: schedule["classes"][k]["room"],
+                period: days[0][day][school][currentSlot]["period"],
+                time: ""};
 
-            default:
-              scheduleObj = {
-                name: days[0][day][school][currentSlot]["period"],
+            } else {
```

Comment:
do you need this else? seems like much of the object is common with the case above?